demos/demo_2d_north_sea.py

```diff
@@ -0,0 +1,234 @@
+# For the map projection, we use the UTM coordinate system, which subdivides the surface
+# of the Earth into zones and applies a tangent plane approximation within each zone.
+# In our case, UTM zone 30 is the appropriate one.
+
+from thetis import *
+import thetis.coordsys as coordsys
+import thetis.forcing as forcing
+import csv
+import os
+import pyproj
+
+sim_tz = timezone.pytz.utc
+UTM_ZONE30 = pyproj.Proj(proj="utm", zone=30, datum="WGS84", units="m", errcheck=True)
+trans = pyproj.Transformer.from_crs(UTM_ZONE30.srs, coordsys.LL_WGS84.srs)
+
+# Having imported all the relevant packages, the first thing we need is a mesh of the
+# domain of interest. This part is skipped for the purposes of this demo, but details
+# can be found in the corresponding
+# `example <https://github.com/thetisproject/thetis/tree/master/examples/north_sea>`__
+# in the Thetis source code. The mesh is stored as a GMSH file named `north_sea.msh`
+# and is plotted below.
+#
+# .. figure:: north_sea_mesh.png
+#    :figwidth: 80%
+#    :align: center
+#
+# Note that the boundary segments are given different tags, depending on whether they
+# correspond to open ocean (tag 100) or coasts (tag 200). This is because we impose
+# different boundary conditions in each case.
+#
+# We set up the (UTM) mesh and calculate its latitude-longitude coordinates as follows:
+
+mesh2d = Mesh("north_sea.msh")
+lonlat = Function(mesh2d.coordinates.function_space())
+for i, xy in enumerate(mesh2d.coordinates.dat.data_ro):
+    lonlat.dat.data[i] = trans.transform(*xy)
+
+# With the mesh, we can now move on to set up fields defined upon it. For the
+# bathymetry data, we use the
+# `ETOPO1 data set <https://www.ngdc.noaa.gov/mgg/global>`__.
+# A NetCDF file containing such data for the North Sea can be downloaded from the
+# webpage, stored as `etopo1.nc` and then interpolated onto the unstructured mesh
+# using SciPy. An `interpolate_bathymetry` script is provided in the
+# `example <https://github.com/thetisproject/thetis/tree/master/examples/north_sea>`__
+# in the Thetis source code, which follows the
+# `recommendations in the Firedrake documentation for interpolating data <https://firedrakeproject.org/interpolation.html#interpolation-from-external-data>`__.
+# However, the NetCDF file cannot be included here for
+# copyright reasons, so we insteady provide a HDF5 file containing the data already
+# interpolated onto the mesh. Note that HDF5 files currently have to be saved and
+# loaded using the same number of processors. The bathymetry field was generated by
+# a serial run, so the following will not work in parallel.
+
+P1_2d = get_functionspace(mesh2d, "CG", 1)
+bathymetry_2d = Function(P1_2d, name="Bathymetry")
+with DumbCheckpoint("north_sea_bathymetry", mode=FILE_READ) as h5:
+    h5.load(bathymetry_2d)
+
+# The resulting bathymetry field is plotted below.
+#
+# .. figure:: north_sea_bathymetry.png
```

Comment:
Maybe you could use a log scale colormap in this case or set the bounds to [0, 500] to show the shallow regions.